src/db/write.rs

```diff
@@ -106,16 +116,28 @@ impl<'a> Write<'a> {
     }
 
     pub fn put(&mut self, key: Vec<u8>, val: Vec<u8>) {
+        match self.map.get(&key) {
+            None => self.checksum.add(&key, &val),
+            Some(old_val) => self.checksum.replace(&key, old_val, &val),
+        };
         self.map.put(key, val)
     }
 
+    pub fn del(&mut self, key: Vec<u8>) {
+        let old_val = self.map.get(&key);
+        match old_val {
+            None => {}
+            Some(old_val) => self.checksum.remove(&key, old_val),
```

Comment:
Same thing we can easily have remove return whether the delete found anything -- the backing store knows.